aiodocker/docker.py

```diff
@@ -500,20 +503,25 @@ def _transform_event(self, data):
                 self._transform_event,
                 human_bool(params['stream']),
             )
-            async for data in self.json_stream:
-                await self.channel.publish(data)
+            try:
+                async for data in self.json_stream:
+                    await self.channel.publish(data)
+            finally:
+                await self.json_stream.close()
+                self.json_stream = None
         finally:
             # signal termination to subscribers
             await self.channel.publish(None)
-            try:
-                await self.json_stream.close()
-            except:
-                pass
```

Comment:
@barrachri the error in the `close()` used to be hidden by this try...except: pass